src/confd/src/infix-containers.c

```diff
@@ -191,55 +194,27 @@ static int add(const char *name, struct lyd_node *cif)
 		fprintf(fp, " %s", string);
 
 	fprintf(fp, "\n");
-
-	if (lydx_is_enabled(cif, "manual"))
-		fprintf(fp, "initctl -bnq cond set container:%s\n", name);
-
 	fchmod(fileno(fp), 0700);
 	fclose(fp);
 
+	/* Enable, or update, container -- both trigger setup script. */
 	systemf("initctl -bnq enable container@%s.conf", name);
-
-	/*
-	 * All start scripts must wait for the rest of confd to complete
-	 * before being enqueued to execd, so we postpone it using this
-	 * "inbox" to the post hook.
-	 */
-	writesf(script, "a", "%s", _PATH_INBOX);
+	systemf("initctl -bnq touch container@%s.conf", name);
 
 	return 0;
 }
 
 static int del(const char *name)
```

Comment:
This is not necessarily new behavior in this series, but I think the fact that we treat "no enabled" in the same way as removing the container as the same state might be problematic.

AFAIK, that means we will also remove any associated volumes, which I don't think is what the user (or yours truly) expects.

Reply:
Wow, good catch!  You are absolutely right, this is old buggy behavior.  Reworking to instead disable the container service, keeping any created container and volumes associated with it so they are not auto-pruned.